### PR TITLE
feat: v4-beta.0

### PR DESCRIPTION
**Pull request by WoodNeck**
## Details
First PR of v4

- https://woodneck.github.io/egjs-view360/

### CODE REVIEW

**Review by daybrush — APPROVED** [review]
LGTM

File: src/sprite/PackedSprite.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    const rows = this._rows;
+    const columns = this._columns;
+
+    const row = Math.floor(index / columns);
+    const column = index - row * columns;
+
+    const width = texture.width / columns;
+    const height = texture.height / rows;
+
+    console.log(width, height);
```
[the remark this answers is not in the document]
Comment (WoodNeck) [reply src/sprite/PackedSprite.ts:45]: This file is removed due to SpinViewer removal, thanks for checking though!

File: src/type/event.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * egjs projects are licensed under the MIT license
+ */
+/**
+ * An event that fires when the component is initialized.
+ * This will be called once after `init()` is called.
+ * @event ready
+ * @property A type of the event.
+ * @property An instance of the component that triggered this event.
+ */
+export interface ReadyEvent<T> {
```
[the remark this answers is not in the document]
Comment (WoodNeck) [reply src/type/event.ts:12]: Good idea :)

File: src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    if (checker(array[idx])) {
+      return idx;
+    }
+  }
+
+  return -1;
+};
+
+export const getObjectOption = <T extends boolean | Partial<object>>(val: T): NoBoolean<T> => typeof val === "object" ? val : {} as any;
+export const toBooleanString = (val: boolean) => val ? "true" : "false";
```
[the remark this answers is not in the document]
Comment (WoodNeck) [reply src/utils.ts:138]: That looks less verbose but it seems like it's working fine 👍
I removed this function though...

File: rollup.config.js [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
   "@egjs/component": "eg.Component",
   "@egjs/agent": "eg.agent",
   "@egjs/imready": "eg.ImReady",
-  "gl-matrix": "glMatrix",
-  "promise-polyfill": "Promise"
-}
-const name = "eg.view360";
+  "gl-matrix": "glMatrix"
+};
+const name = "View360";
```
[the remark this answers is not in the document]
Comment (WoodNeck) [reply rollup.config.js:10]: Well, the rollup config isn't working at all at the moment.
I'll fix that in the future update with your comment.

**Pull request closed without merging**